pkg/controller/volume/persistentvolume/pv_controller_test.go

```diff
@@ -678,6 +678,9 @@ func TestAnnealMigrationAnnotations(t *testing.T) {
 }
 
 func TestUpdateFinalizer(t *testing.T) {
+	// This set of tests ensures that protection finalizer is removed when CSI migration is disabled
+	// and PV controller needs to remove finalizers added by the external-provisioner.
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIMigrationGCE, false)()
```

Comment:
Should this also toggle CSIMigtation feature gate?

Reply:
I think it does pass the test for https://github.com/kubernetes/kubernetes/pull/104722 but I don't quite understand why only one test case enables `CSIMigrationGCE?` and for the `CSIMigration` flag, it is enabled now by default in k/k so I think `migratedDriverGates: []featuregate.Feature{features.CSIMigration}` will not do anything

Reply:
This test case is only using gce pd as an example of a migrated driver. It's not testing every single migrated driver we support.

Even though CSIMigration feature gate is on by default, the unit test can exercise turning the feature gate off, hence why we should still explicitly toggle feature gates in the unit tests.

Reply:
Yeah what I mean is perhaps the test should explicitly turn off CSIMigration gate instead of "not turn on" such gate